source/Ship.cpp

```diff
@@ -3391,6 +3414,94 @@ void Ship::SetTargetSystem(const System *system)
 
 
 
+// Persistent targets for mission NPCs.
+const bool Ship::HasTravelDirective() const
+{
+	return !travelDestinations.empty() || destinationSystem;
+}
+
+
+
+const std::map<const Planet *, bool> Ship::GetStopovers() const
+{
+	return travelDestinations;
+}
+
+
+
+const System *Ship::GetDestinationSystem() const
+{
+	return destinationSystem;
+}
+
+
+
+// Returns true for a mission NPC given a travel directive, when it has arrived
+// in one of its specified destination systems.
+const bool Ship::IsSurveying() const
+{
+	return stayingTime > 0 && currentSystem && currentSystem == destinationSystem;
```

Comment:
Stayingtime was set in another PR, right? Idk if the comment above is true since you dont mention that argument in it

Reply:
Stayingtime seems to have just been part of the original PR. It looks like its a holdover from the other part of the code that I'm not implementing, though, so thanks for catching it!